cmd/clusterctl/client/config/client.go

```diff
@@ -102,16 +104,16 @@ func newConfigClient(path string, options ...Option) (*configClient, error) {
 // Reader define the behaviours of a configuration reader.
 type Reader interface {
 	// Init allows to initialize the configuration reader.
-	Init(path string) error
+	Init(ctx context.Context, path string) error
 
 	// Get returns a configuration value of type string.
 	// In case the configuration value does not exists, it returns an error.
-	Get(key string) (string, error)
+	Get(ctx context.Context, key string) (string, error)
```

Comment:
Looks like this ctx is not used in any of the implmentations, isn't it? Can you check if this is safe to drop?

Reply:
With that I think you can slo drop `ctx` from `UnmarshalKey`, isn't it?

Reply:
Yes, plus I also updated `AlterImage` and `NewTemplate`.